### ISSUE TITLE
Asymmetric semantic search with multilingual-MiniLM-L12-v2?

### ISSUE THREAD

**Issue by ErfolgreichCharismatisch** [UKPLab/sentence-transformers#1463]
I have a corpus with 144,491 entries with around 2000 characters each forming phrases in english and german.
My goal is to enter a query like a question or a set of keywords for it to output the best fitting index in the corpus.

I am using sentence-transformers_paraphrase-multilingual-MiniLM-L12-v2 currently with a

> query_embedding = embedder.encode(query, convert_to_tensor=True,batch_size=6).to('cuda')
> cos_scores = util.cos_sim(query_embedding, corpus_embeddings)[0]
> top_results = torch.topk(cos_scores, k=top_k)

This gives reasonable results, but is there a better approach?

I am asking, because this is an asymmetric semantic search, which should use the MSMARCO Models according to your description, yet those are only in english and https://www.sbert.net/examples/training/ms_marco/multilingual/README.html seems unfinished.

Which approach using sbert do you suggest?

**Comment by nreimers** [comment]
Yes, this is currently work in progress. We hope we can start soon the training process.

For German & English, there are some MSMARCO English-German models on the hub:
https://huggingface.co/sentence-transformers/msmarco-distilbert-multilingual-en-de-v2-tmp-trained-scratch

**Comment by ErfolgreichCharismatisch** [comment]
Interesting.

**Comment by nishanthcgit** [comment]
Eagerly awaiting the results of this training!
